standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```diff
@@ -140,8 +156,9 @@ void alterCatalog(String catalogName, Catalog newCatalog)
    * @throws MetaException something went wrong, usually in the database.
    * @throws TException general thrift exception.
    */
-  void dropCatalog(String catName)
-      throws NoSuchObjectException, InvalidOperationException, MetaException, TException;
+  default void dropCatalog(String catName)
```

Comment:
should throw UnsupportedOperationException

Reply:
Done.